src/main/java/org/minions/devfund/bruno/Squarelotron.java

```diff
@@ -0,0 +1,240 @@
+     * @return the new squarelotron
+     */
+    public Squarelotron mainDiagonalFlip(int ring) {
+        Squarelotron flipped = new Squarelotron(size);
+        moveTopRowToSideColumn(ring, flipped);
+        moveSideColumnToBottomRow(ring, flipped);
+        return flipped;
+    }
+
+    /**
+     * Moves the side column to bottom row.
+     *
+     * @param ring is a number to represent what square move.
+     * @param flip is the new squareltron.
+     */
+    private void moveSideColumnToBottomRow(final int ring, final Squarelotron flip) {
+        for (int i = ring; i <= size - ring - 1; i++) {
+            flip.squarelotron[size - ring][i] = squarelotron[i][size - ring];
+            flip.squarelotron[i][size - ring] = squarelotron[size - ring][i];
+        }
+    }
+
+    /**
+     * Moves the top row to side column.
+     *
+     * @param ring is a number to represent what square move.
+     * @param flip is the new squareltron.
+     */
+    private void moveTopRowToSideColumn(final int ring, final Squarelotron flip) {
+        for (int j = ring - 1; j <= size - ring; j++) {
+            flip.squarelotron[ring - 1][j] = squarelotron[j][ring - 1];
+            flip.squarelotron[j][ring - 1] = squarelotron[ring - 1][j];
+        }
+    }
+
+    /**
+     * Rotates the squarelotron to the right.
+     *
+     * @param numberOfTurns Integer that represent the moves number.
+     */
+    public void rotateRight(final int numberOfTurns) {
+        int actualTurns = getActualTurns(numberOfTurns);
+
+        while (actualTurns != 0) {
+            // copy array
+            int[][] copy = getCopySquarelotron();
+
+            // update squarelotron
+            for (int row = 0; row < size; row++) {
+                updateSquarelotron(copy, row);
+            }
+            actualTurns--;
+        }
+    }
+
+    /**
+     * Gets a squarelotron copy.
+     *
+     * @return a squarelotron matrix copy.
+     */
+    private int[][] getCopySquarelotron() {
+        int[][] copy = new int[size][size];
+        for (int index = 0; index < size; index++) {
+            System.arraycopy(squarelotron[index], 0, copy[index], 0, size);
+        }
+        return copy;
+    }
+
+    /**
+     * Update the index row with respective values.
+     *
+     * @param copy squarelotron matrix copy
+     * @param row  index row.
+     */
+    private void updateSquarelotron(final int[][] copy, final int row) {
+        final int colZero = 0;
+        final int colOne = 1;
+        final int colTwo = 2;
+        final int colThree = 3;
+        final int colFour = 4;
+        final int colFive = 5;
+        final int colSix = 6;
+        final int colSeven = 7;
+        final int rowThree = 3;
+        final int rowFour = 4;
+        final int rowFive = 5;
+        final int rowSix = 6;
+        final int rowSeven = 7;
+        int count = size - 1;
+        for (int col = 0; col < size; col++) {
+            switch (row) {
```

Comment:
Instead of having this extensive switch case.
You can call the first method and then the second one as your teammate suggested.